javascript_nodejs/14.nlp-with-dispatch/dialogs/qna/index.js

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+const { QnAMaker } = require('botbuilder-ai');
+
+// this is the LUIS service type entry in the .bot file.
+const QnA_CONFIGURATION = 'sample.qna';
+const QnA_TOP_N = 1;
+const QnA_CONFIDENCE_THRESHOLD = 0.5;
+class QnADialog {
+    /**
+     * 
+     * @param {Object} botConfig bot configuration from .bot file
+     */
+    constructor(botConfig) {
+        if(!botConfig) throw ('Need bot config');
+
+        // add recogizers
+        const qnaConfig = botConfig.findServiceByNameOrId(QnA_CONFIGURATION);
+        if(!qnaConfig || !qnaConfig.kbId) throw (`QnA Maker application information not found in .bot file. Please ensure you have all required QnA Maker applications created and available in the .bot file. See readme.md for additional information\n`);
+        this.qnaRecognizer = new QnAMaker({
+            knowledgeBaseId: qnaConfig.kbId,
+            endpointKey: qnaConfig.endpointKey,
+            host: qnaConfig.hostname
+        });
+    }
+    /**
+     * 
+     * @param {Object} context context object
+     */
+    async onTurn(context) {
+        // make call to Qna Maker to get results
```

Comment:
Make should be capitalized. 
"Qna Maker" -> "QnA Maker"
Consider adding an "." to the end of the sentence.